src/Messages.ts

```diff
@@ -238,16 +262,143 @@ export default class Messages {
+    }
+
+    const [event, query] = eventUrl.replace(/^http:\/\/leanplum\//, '').split('?')
+    const params = new URLSearchParams(query)
+    const { message, context } = iframe.metadata
+
+    switch (event) {
+      case 'loadFinished':
+        iframe.style.visibility = 'visible'
+        iframe.style.left = '0'
+        if (message['HTML Height'] > 0) {
+          const width = message['HTML Width']
+          iframe.style.height = `${message['HTML Height']}px`
+          iframe.style.width = width
+          iframe.style.left = `calc((100% - ${width}) / 2)`
+
+          const anchorProp = message['HTML Align'].toLowerCase()
+          iframe.style[anchorProp] = '0'
+        } else {
+          iframe.style.top = '0'
+        }
+
+        context.track()
+        break
+      case 'track':
+        context.track(
+          params.get('event'),
+          parseFloat(params.get('value')),
+          params.get('parameters'),
+          params.get('info')
+        )
+        break
+      case 'runAction':
+      case 'runTrackedAction':
+        context[`${event}Named`](params.get('action'))
+        // fall through and close
+      case 'close':
+        iframe.metadata = null
+        iframe.parentNode.removeChild(iframe)
+        break
+    }
+  }
+
+  handleMessage(options: RenderOptions): void {
+    if (this._showRichIAM && options.message.__name__ === 'HTML') {
+      this.resolveFiles(options.message)
+      fetch(options.message['Template'])
+        .then(res => res.text())
+        .then(template => this.renderRichInAppMessage(template, options))
+    } else {
+      this.events.emit('showMessage', this.resolveFields(options))
+    }
+  }
+
+  private renderRichInAppMessage(template: string, options: RenderOptions): void {
+    const messageId = options.message.messageId
+    const vars = JSON.stringify(options.message)
+    const iframe = document.createElement('iframe') as MessageFrame
+    iframe.setAttribute('id', `lp-message-${messageId}`)
+    Object.assign(iframe.style, {
+      borderWidth: 0,
+      position: 'fixed',
+      top: '-100%',
+      left: '-100%',
+      width: '100%',
+      height: '100%',
+      visibility: 'hidden',
+    })
+    document.body.appendChild(iframe)
+
+    // pass message info
+    iframe.metadata = options
+    iframe.contentWindow.messageId = messageId
+
+    // content
+    const content = template
+      .replace('##Vars##', vars)
+      // TODO: remove patches once v11 templates are published
+      .replace(
+        'function routeToBridge(',
+        `function routeToBridge(x) {
+          window.parent.Leanplum.processMessageEvent(window.messageId, x);
+        }
+        function oldRouteToBridge(`
+      )
+      .replace('</style>', `
+.rating-icon {
+  cursor: pointer;
+}
+#close-button:hover {
+  cursor: pointer;
+  opacity: .8;
+}
+#submit-button:hover,
+#button-1:hover,
+#button-2:hover {
+  cursor: pointer;
+  background-color: rgba(0,0,0,.1);
+}
+</style>`)
+      // </TODO>
```

Comment:
Note: these should be removed once the in-app templates (1) send messages to the JS SDK, (2) work good with pointer devices.

Reply:
Patched so that the SDK loads v11 templates at the moment.